### PR TITLE
Nojira | @ericbakenhus | Use siteUrl for base URL in SSR

### PR DESCRIPTION
**Pull request by ericbakenhus**
# READY FOR REVIEW

# Summary
- Fall back to siteUrl when location.origin is empty

# Review By (Date)
- ASAP

# Criticality
- High

# Review Tasks

## Setup tasks and/or behavior to test

1. Look at code

### CODE REVIEW

File: src/components/navigation/MegaMenu/megaMenu.js [hunk 1]
```diff
@@ -40,13 +40,16 @@ const MegaMenu = ({ blok: { topLevelLinks }, blok, className }) => {
     }
   };
 
+  const populatedTopLevelLinks =
+    topLevelLinks?.filter((l) => l?.url || l?.cached_url) ?? [];
```
Comment (ericbakenhus) [inline src/components/navigation/MegaMenu/megaMenu.js:44]: Remove any links with empty urls.

@sherakama Does this look like it accomplishes we're trying to do?

File: src/components/navigation/MegaMenu/megaMenuLinkItem.js [hunk 2]
```diff
@@ -3,19 +3,24 @@ import SbEditable from 'storyblok-react';
...
+          classes={styles.menuLink}
+          hasExternalIcon
+          externalIconClasses={styles.externalIcon}
+        >
+          {linkText}
+        </SbLink>
+      </li>
+    </SbEditable>
+  );
+};
```
Comment (ericbakenhus) [inline src/components/navigation/MegaMenu/megaMenuLinkItem.js:24]: Only in `MegaMenuLinkItem` (no top level links/buttons), if there is no `url` or `cached_url` (empty link) then we set `cached_url` to `home` and `linktype` to `story` which `SBLink` expects for home page links.

File: src/components/navigation/MegaMenu/megaMenuTopLevelLinkItem.js [hunk 3]
```diff
@@ -2,20 +2,23 @@ import React from 'react';
...
   const browserUrl = new URL(
     location.pathname.replace(/\/+$/, '') + location.hash,
-    location.origin
+    baseUrl
   );
+
   const linkUrl = link.url || link.cached_url;
   const strippedLink = linkUrl.replace(/\/+$/, '');
-  const linkUrlFull = new URL(strippedLink, location.origin);
+  const linkUrlFull = new URL(strippedLink, baseUrl);
```
Comment (ericbakenhus) [inline src/components/navigation/MegaMenu/megaMenuTopLevelLinkItem.js:21]: Fall back to siteUrl when location.origin is not available (SSR).

**Comment by ericbakenhus** [comment]
@sherakama This should be ready now.

**Review by sherakama — APPROVED** [review]

**Review by sherakama — APPROVED** [review]
TYVM.

**Pull request merged**